### PR TITLE
chunked: various improvements

### PR DESCRIPTION
**Pull request by giuseppe**
some improvements and fixes for issues I've found while playing with docker.io/gscrivano/zstd-chunked:fcos-rawhide and docker.io/gscrivano/zstd-chunked:fcos-latest.

More details in each commit.

Signed-off-by: Giuseppe Scrivano <gscrivan@redhat.com>

### CODE REVIEW

File: pkg/chunked/storage_linux.go [hunk 1]
```diff
@@ -230,21 +230,21 @@ func makeEstargzChunkedDiffer(ctx context.Context, store storage.Store, blobSize
 func copyFileFromOtherLayer(file *internal.FileMetadata, source string, otherFile *internal.FileMetadata, dirfd int, useHardLinks bool) (bool, *os.File, int64, error) {
 	srcDirfd, err := unix.Open(source, unix.O_RDONLY, 0)
 	if err != nil {
-		return false, nil, 0, err
+		return false, nil, 0, errors.Wrapf(err, "open source file %q", source)
```
Comment (rhatdan) [inline pkg/chunked/storage_linux.go:233]: This looks like a stutter.

File: pkg/chunked/storage_linux.go [hunk 2]
```diff
@@ -230,21 +230,21 @@ func makeEstargzChunkedDiffer(ctx context.Context, store storage.Store, blobSize
...
 	}
 	defer unix.Close(srcDirfd)
 
 	srcFile, err := openFileUnderRoot(otherFile.Name, srcDirfd, unix.O_RDONLY, 0)
 	if err != nil {
-		return false, nil, 0, err
+		return false, nil, 0, errors.Wrapf(err, "open source file %q under target rootfs", otherFile.Name)
```
Comment (rhatdan) [inline pkg/chunked/storage_linux.go:239]: I am worried about stutter here as well.  Doesn't this end up calling an unix.Open()? Which will report an error with the file that the open failed on?

File: pkg/chunked/storage_linux.go [hunk 3]
```diff
@@ -526,27 +526,27 @@ func setFileAttrs(file *os.File, mode os.FileMode, metadata *internal.FileMetada
 
 	if err := unix.Fchown(fd, metadata.UID, metadata.GID); err != nil {
 		if !options.IgnoreChownErrors {
-			return err
+			return errors.Wrapf(err, "chown %q to %d:%d", metadata.Name, metadata.UID, metadata.GID)
```
Comment (rhatdan) [inline pkg/chunked/storage_linux.go:572]: Stutter.

File: pkg/chunked/storage_linux.go [hunk 4]
```diff
@@ -828,7 +828,10 @@ func safeSymlink(dirfd int, mode os.FileMode, metadata *internal.FileMetadata, o
 		destDirFd = int(f.Fd())
 	}
 
-	return unix.Symlinkat(metadata.Linkname, destDirFd, destBase)
+	if err := unix.Symlinkat(metadata.Linkname, destDirFd, destBase); err != nil {
+		return errors.Wrapf(err, "create symlink %q pointing to %q", metadata.Name, metadata.Linkname)
```
Comment (rhatdan) [inline pkg/chunked/storage_linux.go:944]: stutter.

**Comment by rhatdan** [comment]
LGTM other then concerns about stutter.

File: pkg/chunked/storage_linux.go (hunk 4 above)
Comment (giuseppe) [reply pkg/chunked/storage_linux.go:944]: without the extra information, we only get the UNIX error without any pointer to what symlink we were creating

File: pkg/chunked/storage_linux.go (hunk 2 above)
Comment (giuseppe) [reply pkg/chunked/storage_linux.go:239]: no, it won't use `unix.Open` but call directly the openat2 wrapper, that only produces the UNIX error without any additional information.

File: pkg/chunked/storage_linux.go (hunk 1 above)
Comment (giuseppe) [reply pkg/chunked/storage_linux.go:233]: the `unix.Open` call doesn't include any extra information except the errno

**Review by rhatdan — APPROVED** [review]
Ok I fine then.
LGTM

**Comment by rhatdan** [comment]
@flouthoc @nalind @vrothberg @saschagrunert @umohnani8 PTAL

File: pkg/chunked/storage_linux.go [hunk 5]
```diff
@@ -89,7 +89,7 @@ func copyFileContent(srcFd int, destFile string, dirfd int, mode os.FileMode, us
 	src := fmt.Sprintf("/proc/self/fd/%d", srcFd)
 	st, err := os.Stat(src)
 	if err != nil {
-		return nil, -1, err
+		return nil, -1, errors.Wrapf(err, "copy file content for %q", destFile)
```
Comment (vrothberg) [inline pkg/chunked/storage_linux.go:92]: Nit: using `fmt.Errorf` would be more modern

**Review by vrothberg — COMMENTED** [review]
Other than the nit LGTM

**Comment by giuseppe** [comment]
> Other than the nit LGTM

thanks.  Changed to use `fmt.Errorf` instead of `errors.Wrapf`

**Pull request merged**